File: src/content/1.8/code/scala/snippet04.scala
implicit val eitherBifunctor = new Bifunctor[Either] {
  override def bimap[A, B, C, D](f: A => C)(g: B => D): Either[A, B] => Either[C, D] = {
    case Left(x) => Left(f(x))
    case Right(y) => Right(g(y))
  }
}

File: src/content/1.8/code/ocaml/snippet04.ml
module Bifunctor_Either = Bifunctor_From_Bimap (struct
  open Either

  type ('a, 'b) t = ('a, 'b) Either.t

  let bimap f g = function
    | Left a -> Left (f a)
    | Right b -> Right (g b)
end)
